### ISSUE TITLE
3.0.7版本请求的https接口时返回的response.data没有自动解析,还是字符串

### ISSUE THREAD

**Issue by q296494327** [flutterchina/dio#677]
### New Issue Checklist

- [ ] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |      |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps
3.0.7版本请求https接口时返回的response.data没有自动解析为Map,依然还是字符串
而http接口可以正常解析为Map

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.
![1 https的接口](https://user-images.githubusercontent.com/20579810/74631011-ca590f80-5196-11ea-90d8-8e0732c071a1.jpg)
![2 http的接口](https://user-images.githubusercontent.com/20579810/74631016-d1801d80-5196-11ea-89cc-670cd05cfb99.jpg)

**Comment by zhanfu86** [comment]
3.0.8版本也一样
